polyfills/Map/tests.js

```diff
@@ -0,0 +1,183 @@
+var o, generic, callback;
+
+beforeEach(function() {
+	if ('Map' in window) o = new Map();
+	generic = {};
+	callback = function () {};
+})
+
+it("has valid constructor", function () {
+	expect(new Map).to.be.a(Map);
+	expect(new Map()).to.be.a(Map);
+	if ("__proto__" in {}) {
+		expect((new Map).__proto__.isPrototypeOf(new Map())).to.be(true);
+		expect((new Map).__proto__ === Map.prototype).to.be(true);
+	}
+});
+
+it ("can be pre-populated", function() {
+	var a = 1;
+	var b = {};
+	var c = new Map();
+	var m = new Map([[1,1], [b,2], [c, 3]]);
+	expect(m.has(a)).to.be(true);
+	expect(m.has(b)).to.be(true);
+	expect(m.has(c)).to.be(true);
+	expect(m.size).to.equal(3);
+});
+
+it("implements .size()", function () {
+	expect(o.size).to.be(0);
+	o.set("a", "a");
+	expect(o.size).to.be(1);
+	o["delete"]("a"); // Use square-bracket syntax to avoid a reserved word in old browsers
```

Comment:
Should this be a note in the polyfills config.json? I imagine many developers will forget about `delete` being a reserved word even if it is a method on the object being called upon.